experimental/alp/alp/compile_op.py

```diff
@@ -0,0 +1,216 @@
+import os
```

Comment:
I get that you didn't want to use the harness in this first CL but this is really hacky.

Ideally you'd be porting to using the harness later?

One of the reasons I shot for some amount of load bearing logic in python (the minimum possible) is that this is where I expect learning, stats, plotting etc to happen.

I am not interested in reimplementing SciKitLearn, Pandas, Pytorch or other tools that work well out of the box in C++ to be able to experiment.

Reply:
Yes, I agree. But I had major problems with the harness as it is was:
* I wanted to be able to go down to assembly without using the `mlir-c-runner`.
* I wanted to produce the intermediate results in different files
* In each file I wanted to have the `mlir-proto-opt` command to reproduce the IR (starting from linalg)
* I want also to be able to start the transformation pipeline in the middle (this is yet to be implemented, but very easy to do with the hacky harness)
So, yes, this was intended as a quick and hacky way to get these features reusing the main transformations. We are very focused on performance for now, so I need all the IRs in the simplest and quickest way to analyse them and understand what is wrong between one transform and the next. I promise that once we are satisfied with performance we will revisit this.

Reply:
Understood, note that I don't consider any of these to be more than a few minutes to implement in the harness.
You could create a few more starter tasks like #9 and depending on priority we can help address.
You should be able at each step to print in a single file the mlir-proto-opt command embedded with the IR at that level (wherever the pipeline is applied is where you have the command and the IR).

Now your focus should indeed be to accelerate yourself. 
What I am suggesting is you may be hitting diminishing returns of maintaining a custom string-stitchy path.

Reply:
I created:
https://github.com/google/iree-llvm-sandbox/issues/92
https://github.com/google/iree-llvm-sandbox/issues/93
https://github.com/google/iree-llvm-sandbox/issues/94